DataProducts/inc/STMTestBeamHeaders.hh

```diff
@@ -0,0 +1,90 @@
+#ifndef DataProducts_STMTestBeamHeaders_hh
+#define DataProducts_STMTestBeamHeaders_hh
+//
+// Header definitions for STM test beam data
+//
+
+// C++ includes
+#include <iostream>
+#include <vector>
+#include <array>
+#include <ctime>
+#include <iomanip>
+
+namespace mu2e {
+
+  namespace STMTestBeam {
+    struct TriggerHeader {
+      uint16_t deadbeef[2]; // fixed header word (= 0xDEADBEEF)
+      uint16_t datasize[2]; // data size
+      uint16_t n_slices[2]; // number of slices
+      uint16_t trigger_number[2]; // trigger number
+      uint16_t trigger_mode[1]; // trigger mode
+      uint16_t trigger_time[4]; // trigger time
+      uint16_t trigger_offset[2]; // time of first ADC value relative to trigger time
+      uint16_t n_dropped_packets[1]; // number of dropped packets
+
+      uint32_t getFixedHeader() const {
+        return ((uint32_t) deadbeef[1] << 16) | ((uint32_t) deadbeef[0]);
```

Comment:
Thanks, Ray. I've changed to c++-style casting in this commit: d4e60ae8f.

On your other question: this is the format of the data in the binary files so I guess this is more like a fragment than an offline object. This object itself isn't being stored and I'm just using this struct to help read the binary file. Let me know if there is a better place for this than DataProducts (maybe Sources or even the source module itself?)

Reply:
I didn't know it is not written out, that makes the comment less important.  Is it only used in the one module?  If so, it could be specific to that module.  If it is kept separate, but is only used a temp structure in the source module, I'd probably keep it in the Sources dir, but still no strong opinion, just a thought.

Reply:
I think it's better located in the Sources module so I've moved it in this commit: b67531583